sqlserver/datadog_checks/sqlserver/sqlserver.py

```diff
@@ -334,6 +336,14 @@ def _make_metric_list_to_collect(self, custom_metrics):
                 cfg = {'name': name, 'table': table, 'column': column, 'instance_name': db_name, 'tags': tags}
                 metrics_to_collect.append(self.typed_metric(cfg_inst=cfg, table=table, column=column))
 
+        for name, column, metric_fn in DATABASE_FILES_IO:
+            cfg = {'name': name, 'column': column, 'tags': tags}
+            metrics_to_collect.append(
+                SqlVirtualFileIOStats(
+                    cfg, None, lambda *args, **kwargs: metric_fn(self, *args, **kwargs), column, self.log
+                )
```

Comment:
I had to break some of the established patterns of using `typed_metric` because:

1. The data source comes from multiple tables
2. There is another class which declares this table: https://github.com/DataDog/integrations-core/blob/master/sqlserver/datadog_checks/sqlserver/metrics.py#L262, and that causes tests to fail because tables are assumed to be unique here: https://github.com/DataDog/integrations-core/blob/master/sqlserver/datadog_checks/sqlserver/metrics.py#L931-L935 
3. Different columns have different metric types (monotonic_count vs. gauge)